image/daemon/image.go

```diff
@@ -41,15 +41,19 @@ func Image(ref name.Reference) (v1.Image, func(), error) {
 
 	inspect, _, err := c.ImageInspectWithRaw(context.Background(), ref.Name())
 	if err != nil {
+		c.Close()
```

Comment:
What if putting `defer` under the error handling of `NewClientWithOpts`? If we add a new error handling, we don't have to add `c.Close()` again.

```
defer func(){
    if err != nil {
        c.Close()
    }
}
```

Reply:
Good idea, much better than repeating the c.Close for every error case. I adopted the change.